src/PaymentGateways/Gateways/Stripe/StripePaymentElementGateway/StripePaymentElementRepository.php

```diff
@@ -164,7 +165,29 @@ protected function getStripePublishableKey(int $formId): string
      */
     protected function getStripeConnectedAccountKey(int $formId): string
     {
-        return give_stripe_get_connected_account_id($formId);
+        $defaultAccountId = give_stripe_get_connected_account_id($formId);
+
+        $form = DonationForm::find($formId);
+        if (!$form || !$form->blocks) {
+            return $defaultAccountId;
+        }
+
+        $paymentGatewaysBlock = $form->blocks->findByName('givewp/payment-gateways');
+        $gatewaySettings = $paymentGatewaysBlock->getAttribute('gatewaysSettings') ?? [];
+        $stripeSettings = $gatewaySettings['stripe_payment_element'] ?? null;
+
+        if (is_null($stripeSettings) || $stripeSettings['useGlobalDefault'] || is_null(
+                $stripeSettings['accountId'] ?? null
+            )) {
+            return $defaultAccountId;
+        }
+
+        $allStripeAccounts = give_stripe_get_all_accounts();
+        if (!array_key_exists($stripeSettings['accountId'], $allStripeAccounts)) {
+            return $defaultAccountId;
+        }
+
+        return $stripeSettings['accountId'];
```

Comment:
@pauloiankoski i'm going to suggest we do this a little differently...  

If this was a brand new feature for Stripe I think this would be fine (although this is the first time a gateway is referencing the blocks which feels a little weird lol), but since this functionality already exists and we're kind of piggy backing off the existing functionality & still have v2 forms - i'd rather make this fully backwards compatible. So here's what i'm thinking:

Hook into `givewp_form_builder_updated` and use the logic you have here to convert the block attributes to the existing give stripe meta keys referenced in `give_stripe_get_connected_account_id()`

Then, we're really just using the block attributes as an adapter for the existing meta and this function `give_stripe_get_connected_account_id($formId)` would still work.